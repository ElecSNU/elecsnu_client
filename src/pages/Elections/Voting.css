#voting {
    z-index: 100;
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    display: grid;
    place-items: center;
}

#voting-page {
    height: 80%;
    width: 85%;
    display: flex;
    flex-direction: column;
    padding: 1.5% 2.5%;
}
