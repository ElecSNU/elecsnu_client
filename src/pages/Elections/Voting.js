import React, { useEffect, useState } from 'react';
import { toast } from 'react-toastify';
import useLoader from '../../hooks/useLoader';

import ParticleBackground from '../../components/ParticleBackground/ParticleBackground';

import './Voting.css';

const FaceRecog = React.lazy(() => import('./FaceRecog'));
const ErrorPage = React.lazy(() => import('./FaceRecog'));

const VotingPage = () => {
    const showLoader = useLoader();

    // 0 ==> PROCESSING
    // 1 ==> SUCCESSFULL
    // 2 ==> ERROR
    const [faceRecogStatus, setFaceRecogStatus] = useState(
        0
    );

    return (
        <div
            id='voting-page'
            className='background-light-translucent border-radius-15 box-shadow'
        >
            {faceRecogStatus === 0 ? (
                <FaceRecog
                    setFaceRecogStatus={setFaceRecogStatus}
                    faceRecogStatus={faceRecogStatus}
                    showLoader={showLoader}
                />
            ) : faceRecogStatus === 1 ? (
                <></>
            ) : (
                <ErrorPage />
            )}
        </div>
    );
};

const Voting = (props) => {
    const candidates = new URL(
        document.location
    ).searchParams
        .get('candidates')
        .split(',');

    useEffect(() => {
        // toast.info('Please provide camera permissions!');
    }, []);

    return (
        <div id='voting' className='background-light'>
            <ParticleBackground child={VotingPage} />
        </div>
    );
};

export default Voting;
