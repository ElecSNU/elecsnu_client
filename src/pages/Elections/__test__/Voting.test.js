import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Voting from './../Voting'
import { toast } from 'react-toastify';
import React, { useEffect, useState } from 'react';
// import ParticleBackground from '../../components/ParticleBackground/ParticleBackground';


test('Tests the rendering of the Design side', () => {
    const div = document.createElement("div")
    const tree = renderer.create(<index/>).toJSON();
    expect(tree).toMatchSnapshot();
   });