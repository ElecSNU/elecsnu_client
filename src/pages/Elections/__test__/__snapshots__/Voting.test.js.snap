// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the rendering of the Design side 1`] = `<index />`;
